### PR TITLE
Feature/events wip

### PR DESCRIPTION
**Pull request by cryptomental**
##### Description
This PR solves https://github.com/MARKETProtocol/MARKET.js/issues/95

##### Checklist

- [x] Linter status: 100% pass
- [x] Changes don't break existing behavior
- [x] Tests coverage hasn't decreased

##### Affected core subsystem(s)
Contract, Tests

##### Testing
Tested manually and with automated tests

##### Refers/Fixes
Fixes: https://github.com/MARKETProtocol/MARKET.js/issues/95

### CODE REVIEW

**Comment by cryptomental** [comment]
Let's wait for Travis, all tests passed locally before I rebased .

**Comment by cryptomental** [comment]
Not enough collateral, I encountered it once, I will increase it to a safe level.

**Comment by cryptomental** [comment]
I have an impression that OrderFilledCancelledStore.test.ts is fragile due to Travis and clashes between tests when collateral is moved. The tests that I introduced are stable but they require collateral operations and I suspect it affects the test suite set up. I suspect snapshotId = await createEVMSnapshot(web3); should be called as well in the beforeAll(), will check that now.

**Comment by rojosnow** [comment]
@eswarasai Would you review this PR? Thanks!

File: src/contract_wrappers/ContractWrapper.ts [hunk 1]
```diff
@@ -12,8 +12,9 @@ import {
   OrderLib,
   SignedOrder
 } from '@marketprotocol/types';
-import { CollateralEvent, ContractMetaData, MarketError } from '../types';
+
 import { Transaction } from '@0xproject/types';
+import { CollateralEvent, MarketError, OrderFilledEvent } from '../types/index';
```
Comment (eswarasai) [inline src/contract_wrappers/ContractWrapper.ts]: @cryptomental -- You don't need to specify `../types/index` explicitly. `../types` should work just fine. I'm using this has something to do with the IDE?

Comment (cryptomental) [reply src/contract_wrappers/ContractWrapper.ts]: Most likely it was automatically inserted by VScode. I will remove it.

**Comment by cryptomental** [comment]
Thanks @eswarasai ! I modified the import, tests are still green. I hope it is ok to be merged now. FYI @rojosnow .

**Pull request merged**